document-store/src/main/java/org/hypertrace/core/documentstore/GroupBy.java

```diff
@@ -0,0 +1,32 @@
+package org.hypertrace.core.documentstore;
+
+import java.util.List;
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.documentstore.expression.Expression;
+
+@Data
+public class GroupBy {
+  private Expression key; // The key field whose distinct values would be returned
+  private List<GroupingSpec> groupingSpecs; // The list of aggregations required
```

Comment:
Yes, you're right in the sense that these are more like (special) projections.
For Mongo, these need to specified along with the "group" clause. Ref.: https://docs.mongodb.com/upcoming/reference/operator/aggregation/group/.

However, since this interface is not-specific to any DB/implementation, we could consider that. I will try to move it under selections. Doing so might need additional code changes when we increment this library version since selections today are simple strings.

Reply:
+1, should have read this before posting my earlier comment :)